File: backend/test_auth.py
import requests
import json

# Test registration
register_url = "http://localhost:8001/api/auth/register"
login_url = "http://localhost:8001/api/auth/login"

# Registration data
register_data = {
    "email": "test@example.com",
    "password": "testpassword",
    "user_type": "student",
    "full_name": "Test User"
}

print("Testing registration...")
try:
    response = requests.post(register_url, json=register_data)
    print(f"Registration status code: {response.status_code}")
    print(f"Registration response: {response.json()}")
    registration_success = response.status_code == 200
except Exception as e:
    print(f"Registration failed: {e}")
    registration_success = False

# Login data
login_data = {
    "identifier": "test@example.com",
    "password": "testpassword",
    "user_type": "student"
}

if registration_success:
    print("\nTesting login...")
    try:
        response = requests.post(login_url, json=login_data)
        print(f"Login status code: {response.status_code}")
        print(f"Login response: {response.json()}")
        if response.status_code == 200:
            print("Login successful!")
        else:
            print("Login failed!")
    except Exception as e:
        print(f"Login failed: {e}")
else:
    print("Skipping login test due to registration failure")

File: test_auth.js
const axios = require('axios');

// Test the authentication endpoints
async function testAuth() {
  const API_BASE = 'http://localhost:8000/api';
  
  console.log('Testing authentication endpoints...');
  console.log('API Base:', API_BASE);
  
  try {
    // Test login with invalid credentials (should fail)
    console.log('\n--- Testing Login with Invalid Credentials ---');
    const loginResponse = await axios.post(`${API_BASE}/auth/login`, {
      identifier: 'test@example.com',
      password: 'wrongpassword',
      user_type: 'student'
    });
    console.log('Unexpected success:', loginResponse.data);
  } catch (error) {
    console.log('Expected login error:', error.response?.data || error.message);
  }
  
  try {
    // Test signup
    console.log('\n--- Testing Signup ---');
    const signupResponse = await axios.post(`${API_BASE}/auth/register`, {
      email: 'testuser@example.com',
      password: 'testpassword',
      user_type: 'student',
      full_name: 'Test User'
    });
    console.log('Signup response:', signupResponse.data);
    
    // If signup is successful, try to login with the new user
    console.log('\n--- Testing Login with New User ---');
    const loginResponse = await axios.post(`${API_BASE}/auth/login`, {
      identifier: 'testuser@example.com',
      password: 'testpassword',
      user_type: 'student'
    });
    console.log('Login response:', loginResponse.data);
  } catch (error) {
    console.log('Signup/Login error:', error.response?.data || error.message);
  }
}

testAuth();